src/group_theory/semidirect_product.lean

```diff
@@ -167,6 +170,12 @@ begin
   rw [← F.map_mul, inl_left_mul_inr_right],
 end
 
+/-- Two maps out of the semidirect product are equal if they're equal after composition
+  with both `inl` and `inr` -/
+lemma hom_ext {f g : (N ⋊[φ] G) →* H} (hl : f.comp inl = g.comp inl)
+  (hr : f.comp inr = g.comp inr) : f = g :=
```

Comment:
```suggestion
@[ext]
lemma hom_ext {f g : (N ⋊[φ] G) →* H} (hl : f.comp inl = g.comp inl)
  (hr : f.comp inr = g.comp inr) : f = g :=
```
Not strictly necessary, but it would match how the category theory library works.

Reply:
The reason I didn't do this was because it would maybe clash with the attribute already on `monoid_hom`s that says they're equal when they're equal on all elements.